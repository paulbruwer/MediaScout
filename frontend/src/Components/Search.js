import { React, useState } from "react";
import { Container, Row, Form, InputGroup, Button } from "react-bootstrap";
import Results from "./Results";
import { AiFillHeart } from "react-icons/ai";

//search function takes a search term and media category and sends it off to 
// our proxy server to request a list of items from 3rd party website
function Search(){

    const [search, setSearch] = useState(""); //state defines our search term
    const [mediaCheck, setMedia] = useState("software"); //state defines our media category as set by radios
    const [displayMedia, setDisplayMedia] = useState("software"); // also defines media category, but does not change unless search button is clicked. this makes sure the rendered list doesn't update on radio click
    const [refresh, setRefresh] = useState(true); //state is handed down to Results component as prop to force a refresh

    // initiates response from 3rd party server and response.json to be populated
    const getSearch = async(term=search, media=mediaCheck) => {
        try {
            await fetch(`/search?term=${term}&media=${media}`)
            setDisplayMedia(mediaCheck);
            setRefresh(!refresh);
        } catch (error) {
            console.log(error);
        } 
    }

    return (
        <div>
            <Container style={{backgroundColor: "#B9B4C7", borderRadius:"20px", color:"rgb(53, 47, 68)"}}>
                <Row>
                    <h1>HackemTunes</h1>
                </Row>
                <Row>
                    <Container style={{paddingLeft:"10%", paddingRight:"10%"}}>
                        <Form style={{color: "rgb(53, 47, 68, 0.7)"}}>
                            <InputGroup style={{paddingBottom:"3%"}}>
                                <Form.Control placeholder="Search by keyword" onChange={(e)=>setSearch(e.target.value)} />
                                <Button variant="secondary" onClick={()=>getSearch()}>Search</Button>
                                <Button variant="secondary" href="/favorites"><AiFillHeart /></Button>
                            </InputGroup>
                            <Form.Group>
                                <Form.Check
                                    inline
                                    label="App"
                                    type={'radio'}
                                    name="media"
                                    id={'inline-radio-app'}
                                    defaultChecked
                                    onClick={()=>{setMedia("software")}}
                                />
                                <Form.Check
                                    inline
                                    label="iBooks"
                                    type={'radio'}
                                    name="media"
                                    id={'inline-radio-iBooks'}
                                    onClick={()=>setMedia("ebook")}
                                />
                                <Form.Check
                                    inline
                                    label="Movies"
                                    type={'radio'}
                                    name="media"
                                    id={'inline-radio-movies'}
                                    onClick={()=>setMedia("movie")}
                                />
                                <Form.Check
                                    inline
                                    label="Podcast"
                                    type={'radio'}
                                    name="media"
                                    id={'inline-radio-podcast'}
                                    onClick={()=>setMedia("podcast")}
                                />
                                <Form.Check
                                    inline
                                    label="Music"
                                    type={'radio'}
                                    name="media"
                                    id={'inline-radio-music'}
                                    onClick={()=>setMedia("music")}
                                />
                                <Form.Check
                                    inline
                                    label="Music videos"
                                    type={'radio'}
                                    name="media"
                                    id={'inline-radio-musicVideos'}
                                    onClick={()=>setMedia("musicVideo")}
                                />
                                <Form.Check
                                    inline
                                    label="Audio Books"
                                    type={'radio'}
                                    name="media"
                                    id={'inline-radio-audioBooks'}
                                    onClick={()=>setMedia("audiobook")}
                                />
                                <Form.Check
                                    inline
                                    label="TV-shows"
                                    type={'radio'}
                                    name="media"
                                    id={'inline-radio-tvShows'}
                                    onClick={()=>setMedia("tvShow")}
                                />
                                <Form.Check
                                    inline
                                    label="Short Film"
                                    type={'radio'}
                                    name="media"
                                    id={'inline-radio-shortFilm'}
                                    onClick={()=>setMedia("shortFilm")}
                                />
                            </Form.Group>
                        </Form>
                    </Container>
                </Row>
            </Container>
            <Results mediaType={displayMedia} refresh={refresh}/>
        </div>
)}

export default Search