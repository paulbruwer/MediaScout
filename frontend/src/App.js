import "./App.css";
import { React } from "react";
import Search from "./Components/Search";
import Favorites from "./Components/Favorites";
import { Routes, Route } from "react-router-dom";

function App() {
  return (
    <div className="App">
      <Routes>
        <Route exact path="/" element={<Search />} />
        <Route exact path="/favorites" element={<Favorites />} />
      </Routes>
    </div>
  );
}

export default App;
