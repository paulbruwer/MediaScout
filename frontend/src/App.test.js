import { render, screen } from '@testing-library/react';
import App from './App';
import renderer from 'react-test-renderer';
import Search from './Components/Search';


test('renders correctly', () => {
  const tree = renderer
  .create(<Search />)
  .toJSON();
  expect(tree).toMatchSnapshot();
  });
