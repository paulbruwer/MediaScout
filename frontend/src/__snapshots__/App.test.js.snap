// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div>
  <div
    className="container"
    style={
      Object {
        "backgroundColor": "#B9B4C7",
        "borderRadius": "20px",
        "color": "rgb(53, 47, 68)",
      }
    }
  >
    <div
      className="row"
    >
      <h1>
        HackemTunes
      </h1>
    </div>
    <div
      className="row"
    >
      <div
        className="container"
        style={
          Object {
            "paddingLeft": "10%",
            "paddingRight": "10%",
          }
        }
      >
        <form
          className=""
          style={
            Object {
              "color": "rgb(53, 47, 68, 0.7)",
            }
          }
        >
          <div
            className="input-group"
            style={
              Object {
                "paddingBottom": "3%",
              }
            }
          >
            <input
              className="form-control"
              onChange={[Function]}
              placeholder="Search by keyword"
            />
            <button
              className="btn btn-secondary"
              disabled={false}
              onClick={[Function]}
              type="button"
            >
              Search
            </button>
            <a
              className="btn btn-secondary"
              disabled={false}
              href="/favorites"
              onClick={[Function]}
              onKeyDown={[Function]}
              role="button"
              tabIndex={0}
            >
              <svg
                fill="currentColor"
                height="1em"
                stroke="currentColor"
                strokeWidth="0"
                style={
                  Object {
                    "color": undefined,
                  }
                }
                viewBox="0 0 1024 1024"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M923 283.6a260.04 260.04 0 0 0-56.9-82.8 264.4 264.4 0 0 0-84-55.5A265.34 265.34 0 0 0 679.7 125c-49.3 0-97.4 13.5-139.2 39-10 6.1-19.5 12.8-28.5 20.1-9-7.3-18.5-14-28.5-20.1-41.8-25.5-89.9-39-139.2-39-35.5 0-69.9 6.8-102.4 20.3-31.4 13-59.7 31.7-84 55.5a258.44 258.44 0 0 0-56.9 82.8c-13.9 32.3-21 66.6-21 101.9 0 33.3 6.8 68 20.3 103.3 11.3 29.5 27.5 60.1 48.2 91 32.8 48.9 77.9 99.9 133.9 151.6 92.8 85.7 184.7 144.9 188.6 147.3l23.7 15.2c10.5 6.7 24 6.7 34.5 0l23.7-15.2c3.9-2.5 95.7-61.6 188.6-147.3 56-51.7 101.1-102.7 133.9-151.6 20.7-30.9 37-61.5 48.2-91 13.5-35.3 20.3-70 20.3-103.3.1-35.3-7-69.6-20.9-101.9z"
                />
              </svg>
            </a>
          </div>
          <div>
            <div
              className="form-check form-check-inline"
            >
              <input
                className="form-check-input"
                defaultChecked={true}
                disabled={false}
                id="inline-radio-app"
                name="media"
                onClick={[Function]}
                type="radio"
              />
              <label
                className="form-check-label"
                htmlFor="inline-radio-app"
                title=""
              >
                App
              </label>
            </div>
            <div
              className="form-check form-check-inline"
            >
              <input
                className="form-check-input"
                disabled={false}
                id="inline-radio-iBooks"
                name="media"
                onClick={[Function]}
                type="radio"
              />
              <label
                className="form-check-label"
                htmlFor="inline-radio-iBooks"
                title=""
              >
                iBooks
              </label>
            </div>
            <div
              className="form-check form-check-inline"
            >
              <input
                className="form-check-input"
                disabled={false}
                id="inline-radio-movies"
                name="media"
                onClick={[Function]}
                type="radio"
              />
              <label
                className="form-check-label"
                htmlFor="inline-radio-movies"
                title=""
              >
                Movies
              </label>
            </div>
            <div
              className="form-check form-check-inline"
            >
              <input
                className="form-check-input"
                disabled={false}
                id="inline-radio-podcast"
                name="media"
                onClick={[Function]}
                type="radio"
              />
              <label
                className="form-check-label"
                htmlFor="inline-radio-podcast"
                title=""
              >
                Podcast
              </label>
            </div>
            <div
              className="form-check form-check-inline"
            >
              <input
                className="form-check-input"
                disabled={false}
                id="inline-radio-music"
                name="media"
                onClick={[Function]}
                type="radio"
              />
              <label
                className="form-check-label"
                htmlFor="inline-radio-music"
                title=""
              >
                Music
              </label>
            </div>
            <div
              className="form-check form-check-inline"
            >
              <input
                className="form-check-input"
                disabled={false}
                id="inline-radio-musicVideos"
                name="media"
                onClick={[Function]}
                type="radio"
              />
              <label
                className="form-check-label"
                htmlFor="inline-radio-musicVideos"
                title=""
              >
                Music videos
              </label>
            </div>
            <div
              className="form-check form-check-inline"
            >
              <input
                className="form-check-input"
                disabled={false}
                id="inline-radio-audioBooks"
                name="media"
                onClick={[Function]}
                type="radio"
              />
              <label
                className="form-check-label"
                htmlFor="inline-radio-audioBooks"
                title=""
              >
                Audio Books
              </label>
            </div>
            <div
              className="form-check form-check-inline"
            >
              <input
                className="form-check-input"
                disabled={false}
                id="inline-radio-tvShows"
                name="media"
                onClick={[Function]}
                type="radio"
              />
              <label
                className="form-check-label"
                htmlFor="inline-radio-tvShows"
                title=""
              >
                TV-shows
              </label>
            </div>
            <div
              className="form-check form-check-inline"
            >
              <input
                className="form-check-input"
                disabled={false}
                id="inline-radio-shortFilm"
                name="media"
                onClick={[Function]}
                type="radio"
              />
              <label
                className="form-check-label"
                htmlFor="inline-radio-shortFilm"
                title=""
              >
                Short Film
              </label>
            </div>
          </div>
        </form>
      </div>
    </div>
  </div>
  <div
    style={
      Object {
        "paddingTop": "30px",
      }
    }
  >
    <div
      className="container"
    >
      <div
        className="row"
      >
        <div />
      </div>
    </div>
  </div>
</div>
`;
